cmd/zrepl/data_conn.c

```diff
@@ -154,6 +159,13 @@ uzfs_submit_writes(zvol_info_t *zinfo, zvol_io_cmd_t *zio_cmd)
 		    write_hdr->len, &metadata, is_rebuild);
 		if (rc != 0)
 			break;
+		/* Update the highest ionum used for checkpointing */
+		running_ionum = zinfo->running_ionum;
+		while (running_ionum < write_hdr->io_num) {
+			atomic_cas_64(&zinfo->running_ionum, running_ionum,
```

Comment:
atomic_cas_64 seems to be using lock. This can be changed to sync_and_fetch or a lock can be used here itself.